server/src/fixed-asset/asset.service.ts

```
@@ -23,11 +26,73 @@ export class AssetService {
     const createdAsset = await this.assetModel.create({
       ...createAssetDto,
       createdBy: userId,
+      organizationId,
       createdAt: new Date(),
     });
 
     this.logger.log(`Fixed asset '${name}' (ID: ${createdAsset._id}) successfully created by user '${userId}'`)
 
     return createdAsset.toObject();
   }
+
+  async findAll(organizationId: string, query: PaginationqQuery): Promise<{total: number, assets: Asset[]}> {
+    const page = query.page ? +query.page : 1;
+    const limit = query.limit ? +query.limit : 5;
+    const docsToSkip = (page - 1) * limit;
+
+    const filter = { organizationId: { $eq: organizationId } };
+
+    const [ assets, total ] = await Promise.all([
+      this.assetModel
+        .find(filter)
+        .skip(docsToSkip)
+        .limit(limit)
+        .lean()
+        .exec(),
+      this.assetModel.countDocuments(filter),
+    ]);
+
+    if (assets.length === 0) {
+      this.logger.error('No fixed assets found for this user in the organization');
+      throw new NotFoundException('No fixed assets found for this user in the organization');
```

For paginated responses there is mostly 200 OK response with empty content is such a case.
You can throw an exception only when ID value wasn't found

---

Fixed.